lib/chef/provider/user/mac.rb

```diff
@@ -0,0 +1,607 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../../resource"
+require_relative "../../dsl/declare_resource"
+require_relative "../../mixin/shell_out"
+require_relative "../../mixin/which"
+require_relative "../user"
+require_relative "../../resource/user/mac_user"
+
+class Chef
+  class Provider
+    class User
+      # A macOS user provider that is compatible with default TCC restrictions
+      # in macOS >= 10.14. See resource/user/mac_user.rb for complete description
+      # of the mac_user resource and how it differs from the dscl resource used
+      # on previous platforms.
+      class MacUser < Chef::Provider::User
+        include Chef::Mixin::Which
+
+        provides :mac_user
+        provides :user, os: "darwin", platform_version: ">= 10.14"
+
+        attr_accessor :auth_authority, :guid, :user_plist
+
+        DSCL_PROPERTY_MAP = {
+            uid: "dsAttrTypeStandard:UniqueID",
+            guid: "dsAttrTypeStandard:GeneratedUID",
+            gid: "dsAttrTypeStandard:PrimaryGroupID",
+            home: "dsAttrTypeStandard:NFSHomeDirectory",
+            shell: "dsAttrTypeStandard:UserShell",
+            comment: "dsAttrTypeStandard:RealName",
+            password: "dsAttrTypeStandard:Password",
+            auth_authority: "dsAttrTypeStandard:AuthenticationAuthority",
+            shadow_hash: "dsAttrTypeNative:ShadowHashData",
+            group_members: "dsAttrTypeStandard:GroupMembers",
+        }.freeze
+
+        def define_resource_requirements
+          super
+
+          %w{dscl sysadminctl plutil dsimport}.each do |bin|
+            requirements.assert(:all_actions) do |a|
+              a.assertion { which(bin) }
+              a.failure_message(Chef::Exceptions::User, "Cannot find binary '#{bin}' on the system for #{new_resource}!")
+            end
+          end
+
+          requirements.assert(:all_actions) do |a|
+            a.assertion do
+              if new_resource.property_is_set?(:salt)
+                new_resource.property_is_set?(:iterations) && new_resource.password =~ /^[[:xdigit:]]{256}$/
+              else
+                true
+              end
+            end
+            a.failure_message(Chef::Exceptions::User, "Password must be a SALTED-SHA512-PBKDF2 shadow hash entropy when a shadow hash salt is given")
+          end
+        end
+
+        def load_current_resource
+          convert_group_name if new_resource.property_is_set?(:gid)
```

Comment:
looks like this method doesn't exist and you likely moved this to the coercion in the resource but forgot to delete it here?

Reply:
It's inherited from `Chef::Provider::User`. Like I mentioned in the comment above, we should probably only resolve the group name in a single place and I think it's here in the provider. Still, since we're managing the group we shouldn't raise an error if the group isn't resolvable so I've removed that method.